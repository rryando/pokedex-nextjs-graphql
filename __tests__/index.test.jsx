import React from "react";
import { useRouter } from "next/router";
import { render, screen, waitFor, fireEvent } from "@testing-library/react";
import Home from "../pages/index";
import pokemonData from "./mocks/pokemonData.json";

describe("Home", () => {
  it("renders a correctly", async () => {
    const { container } = render(<Home />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders a correctly with pokemonData", async () => {
    const { container } = render(<Home pokemonData={pokemonData} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders a correctly with pokemonData and search", async () => {
    const { container } = render(<Home pokemonData={pokemonData} />);

    fireEvent.change(screen.getByPlaceholderText("Search"), {
      target: { value: "pikachu" },
    });

    expect(container.firstChild).toMatchSnapshot();
  });
});
