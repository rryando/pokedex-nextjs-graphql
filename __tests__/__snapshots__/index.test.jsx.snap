// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders a card as fetch complete 1`] = `
<div
  class="container"
>
  <div
    class="pokemonCardContainer"
  >
    <div
      class="pokemonCardGridWrapper"
    />
  </div>
</div>
`;

exports[`Home renders a correctly 1`] = `
<div
  class="container"
>
  <div
    class="pokemonCardContainer"
  >
    <div
      class="pokemonCardGridWrapper"
    />
  </div>
</div>
`;

exports[`Home renders a correctly with pokemonData 1`] = `
<div
  class="container"
>
  <div
    class="pokemonCardContainer"
  >
    <div
      class="pokemonCardGridWrapper"
    />
  </div>
</div>
`;
